src/main/java/seedu/address/model/event/Event.java

```diff
@@ -0,0 +1,137 @@
+package seedu.address.model.event;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import seedu.address.model.particpant.Participant;
+
+/**
+ * This is an Event class.
+ */
+public class Event implements Comparable<Event> {
+
+    private boolean isDone = false;
+    private ArrayList<Participant> participants = new ArrayList<>();
+    private EventName eventName;
+    private EventDate eventDate;
+    private EventTime eventTime;
+
+    /**
+     * This is the constructors of an Event.
+     *
+     * @param name of the Event.
+     * @param date at which the Event occurs.
+     * @param time of the Event.
+     */
+    public Event(EventName name, EventDate date, EventTime time) {
+        this.eventName = name;
+        this.eventDate = date;
+        this.eventTime = time;
+    }
+
+    /**
+     * This is an overloaded constructor of Event.
+     *
+     * @param name of the Event.
+     * @param date at which the Event occurs.
+     */
+    public Event(EventName name, EventDate date) {
+        this(name, date, new EventTime());
+    }
+
+    public EventName getName() {
+        return this.eventName;
+    }
+
+    public EventDate getDate() {
+        return this.eventDate;
+    }
+
+    public EventTime getTime() {
+        return this.eventTime;
+    }
+
+    public boolean getIsDone() {
+        return this.isDone;
+    }
+
+    public List<Participant> getParticipants() {
+        return Collections.unmodifiableList(this.participants);
+    }
+    /**
+     * Marks the Event as done.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Returns true if both events have the same name.
+     * This defines a weaker notion of equality between two events.
+     */
+    public boolean isSameEvent(Event otherEvent) {
```

Comment:
Not criticising this, I just want to ask what this method is supposed to be used for.

Reply:
Not really, it is more for the UniqueEventList class to prevent duplicates. Yeap, every event might need to have a unique name.